src/coreclr/tools/Common/Compiler/DependencyAnalysis/Target_ARM/ARMEmitter.cs

```diff
@@ -139,15 +139,24 @@ public void EmitLDR(Register destination, Register source, int offset)
 
         // movw  reg, [reloc] & 0x0000FFFF
         // movt  reg, [reloc] & 0xFFFF0000
+        // add   reg, pc
         // reg range: [0..12, LR]
         public void EmitMOV(Register destination, ISymbolNode symbol)
         {
             Debug.Assert(destination >= Register.R0 && (destination <= Register.R12 || destination == TargetRegister.LR));
-            Builder.EmitReloc(symbol, RelocType.IMAGE_REL_BASED_THUMB_MOV32);
+            Builder.EmitReloc(symbol, RelocType.IMAGE_REL_BASED_THUMB_MOV32_PCREL); // 12-byte offset is part of the relocation
             Builder.EmitShort(unchecked((short)0xf240));
             Builder.EmitShort((short)((byte)destination << 8));
             Builder.EmitShort(unchecked((short)0xf2c0));
             Builder.EmitShort((short)((byte)destination << 8));
+            if (destination <= Register.R7)
+            {
+                Builder.EmitShort(unchecked((short)(0x4478u + (byte)destination)));
+            }
+            else
+            {
+                Builder.EmitShort(unchecked((short)(0x44f0u + (byte)destination)));
+            }
```

Comment:
Review note: This affects R2R. It generates the same pattern as the JIT but it's one extra instruction. If you feel like we need to special-case it, I'll try to do it.

Reply:
I think it is fine. This pattern is better for private working set.